resources/lib/ipwww_video.py

```diff
@@ -740,47 +742,41 @@ def Search(search_entered):
-    for provider_url, provider_name in providers:
-        # First we query the available streams from this website
-        if channelname == 's4cpbs':
-            url = 'http://a.files.bbci.co.uk/media/live/manifests/hds/pc/%s/%s.f4m' % (
-                provider_url, channelname)
-        else:
-            url = 'http://a.files.bbci.co.uk/media/live/manifesto/audio_video/simulcast/hds/uk/pc/%s/%s.f4m' % (
-                provider_url, channelname)
-        html = OpenURL(url)
-        # Use regexp to get the different versions using various bitrates
-        match = re.compile('href="(.+?)".+?bitrate="(.+?)"').findall(html)
-        streams_available = []
-        for address, bitrate in match:
-            url = address.replace('f4m', 'm3u8')
-            streams_available.append((int(bitrate), url))
-        streams_available.sort(key=lambda x: x[0], reverse=True)
-        # print streams_available
-        # Play the prefered option
-        if bitrate_selected > 0:
-            match = [x for x in streams_available if (x[0] == stream_bitrates[bitrate_selected])]
-            if len(match) == 0:
-                # Fallback: Use any lower bitrate from any source.
-                match = [x for x in streams_available if (x[0] in range(1, stream_bitrates[bitrate_selected - 1] + 1))]
-                match.sort(key=lambda x: x[0], reverse=True)
-            # print "Selected bitrate is %s"%stream_bitrates[bitrate_selected]
-            # print match
-            # print "Playing %s from %s with bitrate %s"%(name, match[0][1], match [0][0])
-            PlayStream(name, match[0][1], iconimage, '', '')
-        # Play the fastest available stream of the preferred provider
+
+    retlist = []
+    NEW_URL = "http://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/iptv-all/vpid/%s/transferformat/hls?cb=%d" % (channelname, random.randrange(10000,99999)) 
+
+    html = OpenURL(NEW_URL)
+
+    # Parse the different streams and add them as new directory entries.
+    match = re.compile(
+        'media.+?bitrate="(.+?)".+?encoding="(.+?)".+?connection.+?href="(.+?)".+?supplier="(.+?)".+?transferFormat="(.+?)"'
+        ).findall(html)
+    for bitrate, encoding, url, supplier, transfer_format in match:
+        retlist.append((supplier, int(bitrate), url, encoding))
+
+    retlist = sorted(retlist, key=lambda x: x[1], reverse=True)
+
+    for (supplier, bitrate, url, encoding) in retlist:
+
+        if bitrate > 2100:
+            color = 'green'
+        elif bitrate > 1000:
+            color = 'yellow'
+        elif bitrate > 600:
+            color = 'orange'
         else:
-            PlayStream(name, streams_available[0][1], iconimage, '', '')
+            color = 'red'
+        title = name + ' - [I][COLOR %s]%s Kbps[/COLOR] [COLOR white]%s[/COLOR] [COLOR grey]%s[/COLOR][/I]' % (color, bitrate, encoding, supplier)
+        print (bitrate, stream_bitrates[bitrate_selected])
```

Comment:
Please remove prints before the pull request, so we don't have any debug code in a release.